src/idatetime.c

```diff
@@ -143,6 +143,43 @@ SEXP convertDate(SEXP x, SEXP type)
     else if (!strcmp(ctype_str, "yearqtr")) { ctype = YEARQTR; ansint = false; }
     else internal_error(__func__, "invalid type, should have been caught before"); // # nocov
     
+    if (ctype == WEEK) {
+        SEXP ans = PROTECT(allocVector(INTSXP, n));
+        int *ansp = INTEGER(ans);
+
+        SEXP opt = GetOption(install("datatable.week"), R_NilValue);
+        const char *mode = isString(opt) && length(opt) == 1 ? CHAR(STRING_ELT(opt, 0)) : "default";
+
+        bool use_sequential = !strcmp(mode, "sequential");
+        bool use_legacy = !strcmp(mode, "legacy");
+        bool can_warn = !use_sequential && !use_legacy && !week_deprecation_warning_issued;
```

Comment:
Also I'm not sure about this global turning off the warning after first hit, since this make it harder to reproduce the warning.

WDYT @jangorecki @MichaelChirico @tdhock 

e.g. 
``` r
week(as.IDate("2023-01-07"))
week(as.IDate("2023-01-07"))
```

will omit the warning once.

Reply:
Last question now is whether we should throw this warning **every time** in next release until user sets her/his option or only on cases **where `week` will diverge** in the future e.g. `week(as.IDate('2023-01-07'))`